cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -433,10 +426,11 @@ private static <C extends Connectable<C>> void writePhaseTapChanger(C eq, PhaseT
             if (regulatingControlId.isPresent()) {
                 String mode = getPhaseTapChangerRegulationMode(ptc);
                 // Only export the regulating control if mode is valid
-                if (mode != null) {
+                if (mode != null && !regulatingControlsWritten.contains(regulatingControlId.get())) {
```

Comment:
I think you should check before fetching `mode`:
```java
if (regulatingControlId.isPresent() && !regulatingControlsWritten.contains(regulatingControlId.get())
```

Reply:
Yes, I meant to move the `get` method two lines before (like it is done below)

Reply:
Agree, better together. I was thinking of avoiding the check in the regulatingControlsWritten if mode was null, but your solutions is cleaner.